### ISSUE TITLE
Make waku-web browser work more reliably

### ISSUE THREAD

**Issue by oskarth** [vacp2p/waku-web-chat#8]
# Problem

Currently this setup is quite janky with browser. NodeJS and nim-waku generally works. But for browser we often get:

`ICE failed, add a TURN server and see about:webrtc for more details` errors and similar, where browser can't establish a connection/discover the node js version.

# Acceptance criteria

Make browser work more reliably, as well as documented caveats of why things might fail etc.

**Comment by oskarth** [comment]
cc @decanus who might know more about this wrt wrtc etc

**Comment by decanus** [comment]
I in-fact do, so when negotating a webrtc connection you need to have a set of TURN servers. This is essentially a relay if a direct connection cannot be established.

**Comment by oskarth** [comment]
Isn't this what the signaling server is already doing? Just badly

**Comment by decanus** [comment]
No, the signaling server is used to establish connections. The TURN server is a proxy if a direct connection cannot be established. See [Turn](https://webrtcforthecurious.com/docs/03-connecting/#turn).

**Comment by oskarth** [comment]
Ah, for some reason I thought this was provided for by https://github.com/libp2p/js-libp2p-webrtc-star

It seems like it isn't https://github.com/libp2p/js-libp2p-webrtc-star/issues/179

It is also worth noting that the *-star family is not recommended

**Comment by oskarth** [comment]
For some reason I thought TURN was provided for by https://github.com/libp2p/js-libp2p-webrtc-star

It is also worth noting that the *-star family is not recommended

**Comment by decanus** [comment]
@oskarth I wouldn't assume it is, as it would need to be some node / server that can be accessed. Not the browser.

**Comment by oskarth** [comment]
I don't mean the browser, I mean the Node JS bootstrap node. https://github.com/vacp2p/waku-web-chat/tree/master/nodejs/bootstrap
